model/cicd/metrics.yaml

```diff
@@ -0,0 +1,65 @@
+groups:
+  - id: metric.cicd.pipeline.run.duration
+    type: metric
+    metric_name: cicd.pipeline.run.duration
+    brief: 'Duration of a pipeline run grouped by pipeline and result.'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: required
+      - ref: cicd.pipeline.result
+        requirement_level: required
+  - id: metric.cicd.pipeline.run.executing
+    type: metric
+    metric_name: cicd.pipeline.run.executing
+    brief: 'The number of pipeline runs currently executing.'
+    instrument: updowncounter
+    unit: "{pipeline_run}"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: required
+  - id: metric.cicd.queue.latency
+    type: metric
+    metric_name: cicd.queue.latency
+    brief: 'The duration a pipeline run takes from being triggered to the start of execution.'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: recommended
+  - id: metric.cicd.queue.length
+    type: metric
+    metric_name: cicd.queue.length
+    brief: 'The number of pipeline runs waiting for their start of execution.'
+    instrument: updowncounter
```

Comment:
Thinking if an UpDownCounter is the best instrument type for this. Given for a updowncounter, instrumentations will have to decrement once the job leaves the queue and goes to execution. Is this how you thought this metric would be used? Otherwise a Gauge may also work. 

https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#consistent-updowncounter-timeseries

Reply:
> Given for a updowncounter, instrumentations will have to decrement once the job leaves the queue and goes to execution. Is this how you thought this metric would be used? 

Yes, I indeed mean for it to work like that.
The `cicd.queue.length` metric is meant to count the pipeline runs that are about to start but have not yet started execution. Once they start executing they are meant to be decremented from this metric.

I was following https://opentelemetry.io/docs/specs/otel/metrics/supplementary-guidelines/ to determine which instrument type to choose.

For the `cicd.queue.length` metric I thought to measure the queue length and that the values were additive over the `cicd.pipeline.name` attribute. As such an asynchronous UpDownCounter would be used.
Of course counting the jobs added/removed with a synchronous UpDownCounter would work as well.

I did shortly wonder if the `cicd.queue.length` might be redundant since I did define `cicd.queue.latency` as a histogram which would also contain a count of queued pipeline runs. However the histogram would only record the values once they have left the queue and thus the `cicd.queue.length` gives complementary insight to the histogram metric since it counts the pipeline runs which have not yet been recorded by the histogram.